dlc/src/main/scala/org/bitcoins/dlc/BinaryOutcomeDLCClient.scala

```diff
@@ -0,0 +1,1056 @@
+
+    val signedCETF = signedPSBTF.flatMap { signedPSBT =>
+      val txT = signedPSBT.finalizePSBT.flatMap(_.extractTransactionAndValidate)
+      Future.fromTry(txT)
+    }
+
+    (signedCETF, P2WSHWitnessV0(toLocalSPK))
+  }
+
+  /** Constructs Remote's CET given sig*G, the funding tx's UTXOSpendingInfo and payouts */
+  def createCETRemote(
+      sigPubKey: ECPublicKey,
+      payout: CurrencyUnit,
+      remotePayout: CurrencyUnit): Future[
+    (Transaction, PartialSignature, P2WSHWitnessV0)] = {
+    val (cetLocalPrivKey, cetRemotePubKey) = if (sigPubKey == sigPubKeyWin) {
+      (cetWinPrivKey, cetRemoteWinPubKey)
+    } else {
+      (cetLosePrivKey, cetRemoteLosePubKey)
+    }
+
+    val remoteToLocalPubKey =
+      cetRemotePubKey.deriveChildPubKey(UInt32.zero).get.key
+    val localToLocalPrivKey = cetLocalPrivKey.deriveChildPrivKey(UInt32.one).key
+    val toRemotePrivKey = cetLocalPrivKey.deriveChildPrivKey(UInt32(2)).key
+
+    val pubKey = sigPubKey.add(remoteToLocalPubKey)
+
+    val toLocalSPK = P2PKWithTimeoutScriptPubKey(
+      pubKey = pubKey,
+      lockTime = ScriptNumber(timeouts.penaltyTimeout),
+      timeoutPubKey = localToLocalPrivKey.publicKey
+    )
+
+    val toLocal: TransactionOutput =
+      TransactionOutput(remotePayout + toLocalClosingFee,
+                        P2WSHWitnessSPKV0(toLocalSPK))
+    val toRemote: TransactionOutput =
+      TransactionOutput(payout, P2WPKHWitnessSPKV0(toRemotePrivKey.publicKey))
+
+    val outputs: Vector[TransactionOutput] = Vector(toLocal, toRemote)
+
+    val fundingTx = createUnsignedFundingTransaction
+    val fundingTxid = fundingTx.txIdBE
+    val fundingOutPoint = TransactionOutPoint(fundingTxid, UInt32.zero)
+    val fundingInput =
+      TransactionInput(fundingOutPoint, EmptyScriptSignature, sequence)
+
+    val unsignedTx = BaseTransaction(TransactionConstants.validLockVersion,
+                                     Vector(fundingInput),
+                                     outputs,
+                                     timeouts.contractMaturity.toUInt32)
+
+    val sigF = PSBT
+      .fromUnsignedTx(unsignedTx)
+      .addUTXOToInput(fundingTx, index = 0)
+      .addScriptWitnessToInput(P2WSHWitnessV0(fundingSPK), index = 0)
+      .sign(inputIndex = 0, fundingPrivKey)
+      .map(_.inputMaps.head.partialSignatures.head)
+
+    sigF.map((unsignedTx, _, P2WSHWitnessV0(toLocalSPK)))
+  }
+
+  lazy val createUnsignedRefundTx: Transaction = {
+    val fundingTx = createUnsignedFundingTransaction
+    val fundingTxid = fundingTx.txId
+    val fundingInput = TransactionInput(
+      TransactionOutPoint(fundingTxid, UInt32.zero),
+      EmptyScriptSignature,
+      sequence)
+    val fundingOutput = fundingTx.outputs.head
+
+    val (initiatorValue, initiatorKey, otherValue, otherKey) =
+      if (isInitiator) {
+        val toLocalValue = Satoshis(
+          (fundingOutput.value * input).satoshis.toLong / totalInput.satoshis.toLong)
```

Comment:
division always makes me nervous when handling money, I don't see anything wrong here but just something to be wary of. Probably want to make sure that this is tested well

Reply:
Agreed, and this (as well as fee calculations) will make it to the spec eventually. At this point I'd think of it as nearly right but probably still a placeholder